### PR TITLE
Fix: folder default permission

### LINKED ISSUE #225: Escalade de privilèges - droits d'accès enfants GED
Group 27197

On en avait discuté à l'époque mais désormais c'est nécessaire d'empêcher l'utilisateur de pouvoir mettre des droits supérieur au parents à un enfant de ce dernier (Antoine me dit dans l'oreillette que ça s'appelle l'escalade de privilèges). 

Notre solution serait de "griser" et de ne plus pas rendre cliquable dans le select le droit en question. (cf. image ci-dessus)

### PR DESCRIPTION
**Pull request by NicolasRichel**
https://platform-dev-folder-default-permission.bimdata.io

## Description

closes: #225 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have tested my code.
- [x] I want to run the tests for the commits of this PR

### CODE REVIEW

File: src/components/specific/files/folder-access-manager/group-permission-selector/GroupPermissionSelector.vue [hunk 1]
```diff
@@ -64,19 +57,22 @@ export default {
...
-    );
+    const permissionList = computed(() => [
+      { id: "default", value: props.folder.default_permission },
+      { id: "accessDenied", value: FILE_PERMISSION.ACCESS_DENIED },
+      { id: "readOnly", value: FILE_PERMISSION.READ_ONLY },
+      { id: "readWrite", value: FILE_PERMISSION.READ_WRITE }
+    ]);
 
     const groupPermission = ref();
     watch(
```
Comment (Kurtil) [inline src/components/specific/files/folder-access-manager/group-permission-selector/GroupPermissionSelector.vue:72]: Isn't it the best place for a `watchEffect` instead ?

File: src/i18n/lang/fr.json
Comment (Amoki) [inline src/i18n/lang/fr.json:1]: We have to make sure i18n stays is sync. We have new translations with same tokens

Comment (NicolasRichel) [reply src/i18n/lang/fr.json:1]: Translations will be marked as "fuzzy" in POEditor.
I'll fix them when the PR is merged.

File: src/components/specific/files/folder-access-manager/FolderAccessManager.vue [hunk 2]
```diff
@@ -77,29 +81,36 @@ export default {
...
       );
-
-      listFilter.value = useListFilter(
-        currentFolder.value.groups_permissions.map(({ group }) => group),
-        group => group.name
-      );
     };
 
-    onMounted(() => getFolderInfo());
+    onMounted(() => loadFolderInfo());
```
Comment (Kurtil) [inline src/components/specific/files/folder-access-manager/FolderAccessManager.vue]: Is the arrow function useful ?

File: src/state/groups.js [hunk 3]
```diff
@@ -88,13 +88,8 @@ const updateGroupMembers = async (project, group, members) => {
...
 
-const updateGroupPermission = async (project, folder, group, permission) => {
-  return await GroupService.updateGroupPermission(
-    project,
-    folder,
-    group,
-    permission
-  );
+const updateGroupPermission = async (project, folder, group, data) => {
+  return await GroupService.updateGroupPermission(project, folder, group, data);
```
Comment (Kurtil) [inline src/state/groups.js]: `async/await` may not be needed here.

**Review by Kurtil — CHANGES_REQUESTED** [review]

File: src/state/groups.js (hunk 3 above)
Comment (Amoki) [reply src/state/groups.js]:
```javascript
const updateGroupPermission = GroupService.updateGroupPermission;
```
It does the same, no?

Comment (NicolasRichel) [reply src/state/groups.js]: yep

File: src/components/specific/files/folder-access-manager/FolderAccessManager.vue (hunk 2 above)
Comment (NicolasRichel) [reply src/components/specific/files/folder-access-manager/FolderAccessManager.vue]: I don't think so.

**Pull request merged**